document.go

```diff
@@ -356,6 +328,47 @@ func (d Document) Flag(flag DocumentFlag) bool {
 	return d.data.flag.Is(flag)
 }
 
+// Adds a prefix to field names
+func prefixFieldNames(fieldNames []string, prefix string) []string {
+	if prefix == "" {
+		return fieldNames
+	}
+	// this copy is necessary as embedded structs can be reused
+	newNames := make([]string, len(fieldNames))
+	for i, name := range fieldNames {
+		newNames[i] = prefix + name
+	}
+	return newNames
+}
+
+// Adds a field index and checks for conflicts
+func (d *documentData) addFieldIndex(name string, index []int) {
+	if _, ok := d.index[name]; ok {
+		panic("rel: conflicting field (" + name + ") in struct")
+	}
+	d.index[name] = index
+}
+
+// Transfer values from other document data
+func (d *documentData) mergeEmbedded(other documentData, indexPrefix int, namePrefix string) {
+	for name, path := range other.index {
+		d.addFieldIndex(namePrefix+name, append([]int{indexPrefix}, path...))
+	}
+	appendWithPrefix := func(slice, newNames []string) []string {
```

Comment:
maybe this function can be extracted?

Reply:
Should it really? It's here only to reduce repetitiveness by capturing the name prefix

Reply:
I prefer it that way rather then creating anonymous function
or alternatively this can be merged with `prefixFieldNames` as well 🤔